src/extension/inpage-script/index.js

```diff
@@ -4,87 +4,109 @@ import LiquidProvider from "~/extension/providers/liquid";
 import NostrProvider from "~/extension/providers/nostr";
 import WebBTCProvider from "~/extension/providers/webbtc";
 import WebLNProvider from "~/extension/providers/webln";
+import shouldInjectInpage from "./shouldInject";
 
-if (document) {
-  // Intercept any `lightning:` requests
+function init() {
+  if (document) {
```

Comment:
This document check is in shouldInjectInpage already, right? Remove duplicate check.

Reply:
```
export function doctypeCheck() {
  const doctype = window.document.doctype;
  if (doctype) {
    return doctype.name === "html";
  } else {
    return true;
  }
}
```

does window.document will always exist? i don't know much in this context